api/app/weather_models/machine_learning.py

```diff
@@ -171,6 +174,21 @@ def learn(self):
         self.regression_models_v2.collect_data(query)
         self.regression_models_v2.train()
 
+        # Precip is based on 24 hour periods at 20:00 hours UTC. Use the start_date
```

Comment:
I think we can encapsulate this chunk of logic into it's own class and have it managed by `RegressionModelsV2` similar what was done for `WindDirectionModel`.

Reply:
I could pull it out into a separate function, I'm not sure it makes sense to throw all of it into its own class though? I think I still need to query for the data using `self.session', unless I pass the session to my class.
I can add the `PrecipModel` to `RegressionModelV2' though.

Reply:
Yeah true, I think the interface I introduce doesn't generalize to precip -- it's too different. Feel free to disregard this comment.